Tone/source/ExternalInput.js

```diff
@@ -0,0 +1,220 @@
+define(["Tone/core/Tone", "Tone/source/Source"], function(Tone){
+
+	"use strict";
+
+	/**
+	 *  @class  Tone.ExternalInput is a WebRTC Audio Input. Check 
+	 *          [Media Stream API Support](https://developer.mozilla.org/en-US/docs/Web/API/MediaStream_API)
+	 *          to see which browsers are supported. As of
+	 *          writing this, Chrome, Firefox, and Opera 
+	 *          support Media Stream. Chrome allows enumeration 
+	 *          of the sources, and access to device name over a 
+	 *          secure (HTTPS) connection. See [https://simpl.info](https://simpl.info/getusermedia/sources/index.html) 
+	 *          vs [http://simple.info](https://simpl.info/getusermedia/sources/index.html) 
+	 *          on a Chrome browser for the difference.
+	 *         
+	 *  @constructor
+	 *  @extends {Tone.Source}
+	 *  @param {number} [inputNum=0] If multiple inputs are present, select the input number. Chrome only.
+	 *  @example
+	 *  var motu = new Tone.ExternalInput(3);
+	 *  var soundflower = new Tone.ExternalInput().connect()
+	 *  
+	 *  Tone.ExternalInput.onload = function(){
+	 *  	motu.start();
+	 *  	selectSoundflower(Tone.ExternalInput.sourceList);
+	 *  }
+	 *  	
+	 *  function selectSoundflower(sources){
+	 *  	for(var i = 0; i < sources.length; i++){
+	 *  		if(sources[i].label === "soundflower"){
+	 *  			soundflowerIn.inputNum = i;
+	 *  			soundflowerIn.start();
+	 *  		}
+	 *  	}
+	 *  };
+	 */
+
+	Tone.ExternalInput = function(){
+
+		var options = this.optionsObject(arguments, ["inputNum", "onload"], Tone.ExternalInput.defaults);
+		Tone.Source.call(this, options);
+
+		/**
+		 *  @type {MediaStreamAudioSourceNode}
+		 *  @private
+		 */
+		this._mediaStream = null;
+		
+		/**
+		 *  @type {LocalMediaStream}
+		 *  @private
+		 */
+		this._stream = null;
+		
+		/**
+		 *  @type {Object}
+		 *  @private
+		 */
+		this._constraints = {"audio" : true};
+
+		if(Tone.ExternalInput.sourceList.length === 0){
+			Tone.ExternalInput._getSources();
+		} 
+
+		/**
+		 * The input source position in Tone.ExternalInput.sourceList. 
+		 * Set before ExternalInput.start().
+		 * @type {[type]}
+		 */
+		this.inputNum = options.inputNum;
+
+	};
+
+	Tone.extend(Tone.ExternalInput, Tone.Source);
+
+	/**
+	 * the default parameters
+	 * @type {Object}
+	 */
+	Tone.ExternalInput.defaults = {
+		"inputNum" : 0,
+		"onload" : Tone.noOp
+	};
+
+	/**
+	 * Start the stream
+	 * @private
+	 */
+	Tone.ExternalInput.prototype._start = function(){
+		if(!Tone.ExternalInput.sourceList.length && Tone.ExternalInput._hasGetSources && this.inputNum){
+			console.error("Default source selected, use ExternalInput.onload to allow source loading");
```

Comment:
Now I'm thinking it may be best to remove the errors. If the user is on Mozilla they can select the input they want through the browser interface (this._constraints.audio.optional.sourceId does nothing) and if they're on Chrome, they can use Tone to select the inputNum.